File: src/test/java/org/panteleyev/jpackage/JPackageGradlePluginTest.java
// Copyright © 2022-2026 Petr Panteleyev
// SPDX-License-Identifier: BSD-2-Clause
package org.panteleyev.jpackage;

import org.gradle.testfixtures.ProjectBuilder;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;

public class JPackageGradlePluginTest {
    @Test
    public void pluginRegistersTask() {
        var project = ProjectBuilder.builder().build();
        project.getPlugins().apply("org.panteleyev.jpackageplugin");
        assertNotNull(project.getTasks().findByName("jpackage"));
    }
}

File: src/test/kotlin/org/panteleyev/jpackage/JPackageGradlePluginTest.kt
/*
 Copyright (c) Petr Panteleyev. All rights reserved.
 Licensed under the BSD license. See LICENSE file in the project root for full license information.
 */
package org.panteleyev.jpackage

import org.gradle.testfixtures.ProjectBuilder
import org.testng.Assert.assertNotNull
import org.testng.annotations.Test

class JPackageGradlePluginTest {
    @Test
    fun `plugin registers task`() {
        val project = ProjectBuilder.builder().build()
        project.plugins.apply("org.panteleyev.jpackageplugin")

        assertNotNull(project.tasks.findByName("jpackage"))
    }
}
